src/components/chip/chip.tsx

```diff
@@ -0,0 +1,154 @@
+import { Component, Element, h, Prop } from '@stencil/core';
+import { Icon, Link } from '../../interface';
+import { getIconName } from '../icon/get-icon-props';
+import {
+    makeEnterClickable,
+    removeEnterClickable,
+} from '../../util/make-enter-clickable';
+
+/**
+ * Chips and buttons are both interactive elements in UI design,
+ * but they serve different purposes and are used in different contexts.
+ *
+ * :::warning
+ * Do not use the chip component, carelessly as an alternative for
+ * [`limel-button`](#/component/limel-button/) in the UI design!
+ * In short, buttons are used to trigger actions,
+ * while chips are used to represent entities or choices.
+ *
+ * **Buttons:**
+ * Buttons are used to trigger actions. They are typically used to
+ * submit forms, open dialogs, initiate a process, or perform any action
+ * that changes the state of the application.
+ * Buttons' labels usually contain action words, in other words, the labels is
+ * a _verb in imperative mood_ such as "Submit" or "Delete".
+ * Buttons are placed in areas where it's clear they will initiate
+ * an action when clicked.
+ *
+ * **Chips:**
+ * Chips are compact elements that represent an input, attribute, or action;
+ * often representing choices, filters, or tags, in a small block
+ * or clearly bundled into a group.
+ * :::
+ *
+ * @private
+ * @exampleComponent limel-example-chip-button
+ * @exampleComponent limel-example-chip-link
+ * @exampleComponent limel-example-chip-icon-colors
+ */
+@Component({
+    tag: 'limel-chip',
+    shadow: true,
+    styleUrl: 'chip.scss',
+})
+export class Chip {
+    /**
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
```

Comment:
maybe this should be an optional prop instead

Reply:
That seems reasonable to me.

`id` is not a great name when it's an `@Prop`, since it will conflict with the native `id` attribute. So, if we don't switch to accepting an object of type `Chip`, we should rename this property.

Reply:
Do we even need this? What's this `value?: T;` that the chip.types.ts has? Should it be that one instead? 

Note that we both `id` there too!
```ts
import { Icon } from '../../interface';

export interface Chip<T = any> {
    /**
     * ID of the chip. Must be unique.
     */
    id: number | string;

// ···

    /**
     * Value of the chip.
     */
    value?: T;

// ···
}

```

Reply:
> Do we even need this? 

I'm not entirely sure what we use the id for. I guess we need to check the code.

> What's this `value?: T;` that the chip.types.ts has?

That's a "[generic](https://www.typescriptlang.org/docs/handbook/2/generics.html)". What that does is that it lets the _consumer_ specify what type the `value` property should have. So I could use it like this:

```ts
const myChip: Chip<string> = {
    id: 123,
    value: 'hello',
}
```

Or like this:

```ts
const myChip: Chip<LimeObject> = {
    id: 123,
    value: myCompany,
}
```

> Should it be that one instead?

Not sure what you're asking here. If `id` should have type `T`?

> Note that we both `id` there too!

Yes, but the `id` on `Chip` is never used an an attribute on an html tag. It's perfectly legal to have a property called `id` in javascript, the problem is that since Stencil connects all `@Prop` decorated properties to the correspondingly named html attribute, we get a conflict with the native `id` html attribute.

Reply:
> So, if we don't switch to accepting an object of type Chip, we should rename this property.

Can't we just remove it?

Reply:
Yeah, since we're not using it inside the component, let's just remove it 👍

If we need it for the chip-set or something, we can deal with that then, probably without having to change this component.

Reply:
⚡